core/DbHelper.php

```diff
@@ -177,6 +177,21 @@ public static function createDatabase($dbName = null)
         Schema::getInstance()->createDatabase($dbName);
     }
 
+    /**
+     * Returns if the given table has an index with the given name
+     *
+     * @param string $table
+     * @param string $indexName
+     *
+     * @return bool
+     * @throws Exception
+     */
+    public static function tableHasIndex($table, $indexName)
+    {
+        $result = Db::getReader()->fetchOne('SHOW INDEX FROM '.$table.' WHERE Key_name = ?', [$indexName]);
```

Comment:
Should probably use `Db::get()` here since the `log_visit` queries go to the master and not the slave. And some versions allow you to define different indexes on master vs reader (which is really cool btw since you need different indexes usually on master and reader)